web/packages/teleport/src/Console/DocumentDb/ConnectDialog.tsx

```diff
@@ -0,0 +1,209 @@
+
+import { useTeleport } from 'teleport';
+import { Database } from 'teleport/services/databases';
+import { DbConnectData } from 'teleport/lib/term/tty';
+
+export function ConnectDialog(props: {
+  clusterId: string;
+  serviceName: string;
+  onClose(): void;
+  onConnect(data: DbConnectData): void;
+}) {
+  // Fetch database information to pre-fill the connection parameters.
+  const ctx = useTeleport();
+  const [attempt, getDatabase] = useAsync(
+    useCallback(async () => {
+      const response = await ctx.resourceService.fetchUnifiedResources(
+        props.clusterId,
+        {
+          query: `name == "${props.serviceName}"`,
+          kinds: ['db'],
+          sort: { fieldName: 'name', dir: 'ASC' },
+          limit: 1,
+        }
+      );
+
+      // TODO(gabrielcorado): Handle scenarios where there is conflict on the name.
+      if (response.agents.length !== 1 || response.agents[0].kind !== 'db') {
+        throw new Error('Unable to retrieve database information.');
+      }
+
+      return response.agents[0];
+    }, [props.clusterId, ctx.resourceService, props.serviceName])
+  );
+
+  useEffect(() => {
+    void getDatabase();
+  }, [getDatabase]);
+
+  return (
+    <Dialog
+      dialogCss={dialogCss}
+      disableEscapeKeyDown={false}
+      onClose={props.onClose}
+      open={true}
+    >
+      <DialogHeader mb={4}>
+        <DialogTitle>Connect To Database</DialogTitle>
+      </DialogHeader>
+
+      {attempt.status === 'error' && <Danger children={attempt.statusText} />}
+      {(attempt.status === '' || attempt.status === 'processing') && (
+        <Box textAlign="center" m={10}>
+          <Indicator />
+        </Box>
+      )}
+      {attempt.status === 'success' && (
+        <ConnectForm
+          db={attempt.data}
+          onConnect={props.onConnect}
+          onClose={props.onClose}
+        />
+      )}
+    </Dialog>
+  );
+}
+
+function ConnectForm(props: {
+  db: Database;
+  onConnect(data: DbConnectData): void;
+  onClose(): void;
+}) {
+  const dbUserOpts = props.db.users?.map(user => ({
+    value: user,
+    label: user,
+  }));
+  const dbNamesOpts = props.db.names?.map(name => ({
+    value: name,
+    label: name,
+  }));
+  const dbRolesOpts = props.db.roles?.map(role => ({
+    value: role,
+    label: role,
+  }));
+
+  const [selectedName, setSelectedName] = useState<Option>(dbNamesOpts?.[0]);
+  const [selectedUser, setSelectedUser] = useState<Option>(dbUserOpts?.[0]);
+  const [selectedRoles, setSelectedRoles] = useState<readonly Option[]>();
+
+  const dbConnect = () => {
+    props.onConnect({
+      serviceName: props.db.name,
+      protocol: props.db.protocol,
+      dbName: selectedName.value,
+      dbUser: selectedUser.value,
+      dbRoles: selectedRoles?.map(role => role.value),
+    });
+  };
+
+  return (
+    <Validation>
+      {({ validator }) => (
+        <form>
+          <DialogContent minHeight="240px" flex="0 0 auto">
+            <FieldSelectCreatable
```

Comment:
When we have a wildcard, the fields will be empty, and users must type it out.

![Screenshot 2024-12-13 at 11 26 10](https://github.com/user-attachments/assets/60f6d2d9-4826-4e8a-9330-82e50b79d0e3)

Setting to disable will prevent them from changing the roles (I'm not sure if this will be desired).

Reply:
i must missed. Where is the logic to remove wildcard from db users?